src/reserve-watch/reserve-watch.js

```diff
@@ -0,0 +1,89 @@
+const {
+  Finding, FindingSeverity, FindingType, getJsonRpcUrl,
+} = require('forta-agent');
+const BigNumber = require('bignumber.js');
+const ethers = require('ethers');
+const RollingMath = require('rolling-math');
+
+const contractAddresses = require('../../contract-addresses.json');
+const { 'reserve-watch': Config } = require('../../agent-config.json');
+
+const { windowSize, standardDeviation: limit } = Config;
+const { LendingPoolAddressProvider, ProtocolDataProvider: DataProvider } = contractAddresses;
+const { abi: DataAbi } = require('../../interfaces/AaveProtocolDataProvider.json');
+const { abi: AddressProviderAbi } = require('../../interfaces/ILendingPoolAddressesProvider.json');
+const { abi: PriceOracleAbi } = require('../../interfaces/IPriceOracle.json');
+
+const provider = new ethers.providers.WebSocketProvider(getJsonRpcUrl());
+const protocolDataProvider = new ethers.Contract(DataProvider, DataAbi, provider);
+const poolAddressProvider = new ethers.Contract(LendingPoolAddressProvider,
+  AddressProviderAbi, provider);
+
+let rollingReservePrices = {};
+
+// helper function to create alerts
+function createAlert(asset, price) {
+  return Finding.fromObject({
+    name: `High AAVE ${asset.symbol} Reserve Price Change`,
+    description: `${asset.symbol} Price: ${ethers.utils.formatEther(price)} ether `,
+    alertId: 'AE-AAVE-RESERVE-PRICE',
+    severity: FindingSeverity.Medium,
+    type: FindingType.Suspicious,
+    everestId: '0xa3d1fd85c0b62fa8bab6b818ffc96b5ec57602b6',
+    metadata: {},
+  });
+}
+
+function provideHandleBlock(RollingMathLib) {
+  // Clear out the RollingMath object for ease of testing
+  rollingReservePrices = {};
+
+  return async function handleBlock(blockEvent) {
+    const findings = [];
+
+    // override block number so we get data from the block in question
+    const override = { blockTag: blockEvent.blockNumber };
+
+    // Get the reserve assets
+    const reserveAssets = await protocolDataProvider.getAllReservesTokens({ ...override });
+
+    // Get address of the price oracle
+    const oracleAddress = await poolAddressProvider.getPriceOracle({ ...override });
+    const priceOracle = new ethers.Contract(oracleAddress, PriceOracleAbi, provider);
+
+    await Promise.all(reserveAssets.map(async (asset) => {
```

Comment:
I've had a chance to read up on Promise.all() a little more:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

I don't think it's what we want here or any place where we'll be iterating over tokens, address, etc.  The behavior of Promise.all() is to do one of the following:
- Return upon fulfillment of ALL promises, OR
- Return upon the first rejection of any promise, OR
- Return immediately if the iterable (in this case, the Array returned from calling .map()) is empty

I see two straightforward possibilities for how to fix this:
- Use try...catch inside the function to avoid rejection due to failed contract method calls, math issues, etc.  This should just result in nothing being pushed to `findings` for that particular promise
- Use .map() to create an array of promises and attach .then().catch() to each one, where the .then() would add an entry to `findings` and the .catch() would not (perhaps logging?)